spec/system/admin/bulk_order_management_spec.rb

```diff
@@ -59,6 +59,57 @@
       end
     end
 
+    context "pagination" do
+      let!(:o1) {
+        create(:order_with_distributor, state: 'complete', shipment_state: 'ready',
+                                        completed_at: Time.zone.now )
+      }
+      let!(:o2) {
+        create(:order_with_distributor, state: 'complete', shipment_state: 'ready',
+                                        completed_at: Time.zone.now )
+      }
+
+      before do
+        15.times {
+          create(:line_item_with_shipment, order: o1)
+        }
+        5.times {
+          create(:line_item_with_shipment, order: o2)
+        }
+      end
+
+      it "splits results according to line items from orders" do
+        visit_bulk_order_management
+
+        expect(page).to have_select2 "autogen4", selected: "15 per page" # should be default option
```

Comment:
Where does "autogen4" come from? Could that change when we edit other parts on the page? Is there a better identifier like a label?

**Update:** I just checked and our matcher requires an id while our code doesn't provide one. So we have to do with this auto-generated id for now. Fortunately, we are replacing AngularJS and select2 and this doesn't need fixing.

Reply:
> Ok, so I guess when we do, these tests will need to be updated?

Yes, that's the downside of select2 because it needs special Rspec matchers. If it would work with the normal `select` matchers then we wouldn't need to change it. The new library, tom-select does work with the default matchers. So if we replace it at some point in the future, the specs hopefully don't need changing.

> Sometimes I feel page elements need more consistent and more explicit labeling/identifiers for this type of testing; is also influences Matomo tracking (element clicking, for ex.).

I agree that we could have more labels. In this case, we could also use the text next to it as label which says about how many are display per page. Then you would be able to click on that to open the select input.